File: BACKEND/admin/controllers/systemController.js
const systemService = require("../services/systemService");

function getHealth({ response, sendJson }) {
  sendJson(response, 200, {
    success: true,
    data: systemService.getHealth(),
  });
}

async function getDashboard({ response, sendJson }) {
  sendJson(response, 200, {
    success: true,
    data: await systemService.getDashboard(),
  });
}

module.exports = {
  getHealth,
  getDashboard,
};

File: BACKEND/admin/Controllers/SystemController.cs
using Microsoft.AspNetCore.Mvc;
using MyPuppy.Admin.Models;
using MyPuppy.Admin.Services;

namespace MyPuppy.Admin.Controllers;

[ApiController]
[Route("api/admin")]
public class SystemController : ControllerBase
{
    private readonly SystemService _systemService;

    public SystemController(SystemService systemService)
    {
        _systemService = systemService;
    }

    [HttpGet("health")]
    public IActionResult GetHealth()
        => Ok(ApiResponse<object>.Ok(_systemService.GetHealth()));

    [HttpGet("dashboard")]
    public async Task<IActionResult> GetDashboard()
        => Ok(ApiResponse<object>.Ok(await _systemService.GetDashboardAsync()));
}
